presc/report/runner.py

```diff
@@ -0,0 +1,296 @@
+        """Runs the PRESC report for the given modeling inputs.
+
+        The report is written to <output_path>/presc_report. If this dir already
+        exists, it will be overwritten.
+
+        model: a pre-trained ClassificationModel instance to evaluate
+        test_dataset: a test Dataset instance used to evaluate model performance
+        config_filepath: path to a YAML file listing config options to govern
+            the report behaviour. If specified, this overrides the current
+            config.
+        clean: should previous outputs be cleaned? Default: True
+        """
+        if config_filepath:
+            config = load_config(config_filepath)
+        else:
+            config = self.config
+
+        if clean:
+            self.clean()
+
+        tmpdir = None
+        exec_path = None
+        if self.execution_path:
+            # If using a user-defined execution path, need to make sure
+            # it doesn't exist for `shutil.copytree` to work.
+            # Note that this will only remove the nested subdir, not the actual
+            # user-specified dir.
+            if self.execution_path.exists():
+                shutil.rmtree(self.execution_path)
+            exec_path = self.execution_path
+        else:
+            # Create a temp dir to run the build from.
+            # We set up the temp dir here rather than using jupyter-book's
+            # `run_in_temp` option so that we have access to the temp path.
+            tmpdir = TemporaryDirectory()
+            exec_path = Path(tmpdir.name) / REPORT_EXECUTION_DIR
+
+        # Copy the report source files to the execution dir and
+        # execute from there. The data store for the inputs is saved to
+        # the same dir. That way, since the notebooks' working dir on execution
+        # is set to where they are located by jupyter-book, they can find
+        # the data store without needed to know the calling path.
+        try:
+            shutil.copytree(REPORT_SOURCE_PATH, exec_path)
+        except shutil.Error as e:
+            msg = f"Failed to copy report source to execution dir {exec_path}"
+            raise PrescError(msg) from e
+        # Write the inputs to the data store.
+        ctx = Context(store_dir=exec_path)
+        ctx.store_inputs(model=model, test_dataset=test_dataset, config=config)
+
+        # Build the report.
+        self._run_jb_build(exec_path)
+        if tmpdir:
+            tmpdir.cleanup()
+
+        # The build should have created index.html at the `report_main_page`
+        # path.
+        if self.report_main_page.exists():
+            # Symlink the main page to the top level for convenience.
+            try:
+                main_page_target = self.report_main_page.relative_to(
+                    self._linked_main_page.parent
+                )
+                self._linked_main_page.symlink_to(main_page_target)
+            except OSError:
+                pass
```

Comment:
The `report_html` accessor will give the symlink if it exists, otherwise the target report page. I thought that would be sufficient as the user would be working from Python anyway. This is not yet really documented but it will be 🙂. Not really sure how useful the symlink thing will be, I was just thinking to add some sugar to make it a little more user-friendly.